### DISCUSSION TITLE
My friends can't connect to my TShock server

### DISCUSSION THREAD

**Discussion by Jesse33Max** [Pryaxis/TShock#2913] [Q&A]
I run TShock on my Ubuntu server. I launch the server and then I can connect from my home network, but my friends can't. Ive opened up port 7777 on my router. I also use TailScale VPN on my server so I can acces my server outside my home network. Is there something I missed so that my friends can connect? They tried the ip of my server as wel as the VPN ip of the server.

**Comment by sgkoishi** [comment]
Most server owners don't need a VPN to connect to their servers. If you need one, it's likely your server is not open to the public and nobody can connect to it (without your VPN, and so do your friends).

**Reply by Jesse33Max** [reply]
How else should I ssh into my server outside my home network?

**Reply by sgkoishi** [reply]
There's no difference between your ssh and players connections. If you can ssh to the server without any other tool, so do your players; if you can't and need a VPN, so do your players.

You'll need a public IP address. If not, cloud hosting or port forwarding as an alternative.

**Comment by pcider** [comment]
Make sure your router is accessible via the Internet (your router could be behind another layer of NAT, like CG-NAT). Also make sure you can connect via your public IP (ask google for that, search "my ip"). 

If all else fails, try ngrok tcp if you're just planning to play with your friends

**Reply by Jesse33Max** [reply]
If I use the site https://www.whatismyip.com/ is that ip the same for my server thats on the same network?

**Reply by Jesse33Max** [reply]
https://canyouseeme.org/ I also used this site and it says that port 7777 is closed, but I did open it in my router settings

**Reply by pcider** [reply]
Assuming you did the port-forwarding correctly, then router is probably behind a NAT then. (You could check this if your router's **public** IP shown in its settings does not match the public IP found on whatsmyip)

Then your next choice would be to use something like ngrok or Hamachi.
